src/logic/community/communities.ts

```diff
@@ -163,9 +192,36 @@ export async function createCommunityComponent(
         })
       )
 
+      // Filter by active voice chat if requested
+      let filteredCommunities = communitiesWithThumbnailsAndOwnerNames
+      if (options.onlyWithActiveVoiceChat) {
+        const voiceChatStatuses = await Promise.all(
+          communitiesWithThumbnailsAndOwnerNames.map(async (community) => {
+            try {
+              const status = await commsGatekeeper.getCommunityVoiceChatStatus(community.id)
+              return { communityId: community.id, isActive: status?.isActive ?? false }
+            } catch (error) {
+              // If we can't get the status, assume it's not active
+              logger.warn(`Could not get voice chat status for community ${community.id}`, {
+                error: isErrorWithMessage(error) ? error.message : 'Unknown error'
+              })
+              return { communityId: community.id, isActive: false }
+            }
+          })
+        )
+
+        const activeVoiceChatCommunityIds = new Set(
+          voiceChatStatuses.filter((status) => status.isActive).map((status) => status.communityId)
+        )
+
+        filteredCommunities = communitiesWithThumbnailsAndOwnerNames.filter((community) =>
+          activeVoiceChatCommunityIds.has(community.id)
+        )
+      }
```

Comment:
WDYT about moving the common logic of filtering those communities with active voice chat to a new function so we can reuse it in both places?